tests/qos/test_buffer.py

```diff
@@ -223,50 +223,58 @@ def load_test_parameters(duthost):
-        TESTPARAM_HEADROOM_OVERRIDE = vendor_specific_param['headroom-override']
-        TESTPARAM_LOSSLESS_PG = vendor_specific_param['lossless_pg']
-        TESTPARAM_SHARED_HEADROOM_POOL = vendor_specific_param['shared-headroom-pool']
-        TESTPARAM_EXTRA_OVERHEAD = vendor_specific_param['extra_overhead']
-        TESTPARAM_ADMIN_DOWN = vendor_specific_param['admin-down']
-        MAX_SPEED_8LANE_PORT = vendor_specific_param['max_speed_8lane_platform'].get(duthost.facts['platform'])
-
-        # For ingress profile list, we need to check whether the ingress lossy profile exists
-        ingress_lossy_pool = duthost.shell('redis-cli -n 4 keys "BUFFER_POOL|ingress_lossy_pool"')['stdout']
-        if ingress_lossy_pool:
-            ingress_profile_list = TESTPARAM_ADMIN_DOWN.get('BUFFER_PORT_INGRESS_PROFILE_LIST_TABLE')
-            ingress_profile_list.append('[BUFFER_PROFILE_TABLE:ingress_lossy_zero_profile]')
-
-        # 'admin-down' section contains references to buffer profiles
-        # We need to convert the format of the references according to whether table name should be in the reference
-        if not check_qos_db_fv_reference_with_table(duthost):
-            expected_pgs = TESTPARAM_ADMIN_DOWN.get('BUFFER_PG_TABLE')
-            if expected_pgs:
-                new_pgs = {}
-                for pg, profile in expected_pgs.items():
-                    new_pgs[pg] = profile.replace('[BUFFER_PROFILE_TABLE:', '').replace(']', '')
-                TESTPARAM_ADMIN_DOWN['BUFFER_PG_TABLE'] = new_pgs
-
-            expected_queues = TESTPARAM_ADMIN_DOWN.get('BUFFER_QUEUE_TABLE')
-            if expected_queues:
-                new_queues = {}
-                for queue, profile in expected_queues.items():
-                    new_queues[queue] = profile.replace('[BUFFER_PROFILE_TABLE:', '').replace(']', '')
-                TESTPARAM_ADMIN_DOWN['BUFFER_QUEUE_TABLE'] = new_queues
-
-            expected_ingress_profile_list = TESTPARAM_ADMIN_DOWN.get('BUFFER_PORT_INGRESS_PROFILE_LIST_TABLE')
-            if expected_ingress_profile_list:
-                new_list = []
-                for profile in expected_ingress_profile_list:
-                    new_list.append(profile.replace('[BUFFER_PROFILE_TABLE:', '').replace(']', ''))
-                TESTPARAM_ADMIN_DOWN['BUFFER_PORT_INGRESS_PROFILE_LIST_TABLE'] = new_list
-
-            expected_egress_profile_list = TESTPARAM_ADMIN_DOWN.get('BUFFER_PORT_EGRESS_PROFILE_LIST_TABLE')
-            if expected_egress_profile_list:
-                new_list = []
-                for profile in expected_egress_profile_list:
-                    new_list.append(profile.replace('[BUFFER_PROFILE_TABLE:', '').replace(']', ''))
-                TESTPARAM_ADMIN_DOWN['BUFFER_PORT_EGRESS_PROFILE_LIST_TABLE'] = new_list
+        if 'default_cable_length' in vendor_specific_param:
+            DEFAULT_CABLE_LENGTH_LIST = vendor_specific_param['default_cable_length']
```

Comment:
@stephenxs, as far as I remember correctly if you try to get the parameter that is not defined in _tests/qos/files/dynamic_buffer_param.json_ file, the test execution will return the appropriate error that is why I propose such a simple check to avoid a crash. The list of params may be different for a vendor. What do you think?

Reply:
An exception will be thrown if operator `[]` is used with a non-existed member. But no exception for method `get`.
Please check this example:
```
Python 3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a = {1:1}
>>> a[2]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 2
>>> b = a.get(2)
>>> print(b)
None
>>> 
```